Adaptors/MongoDB/src/ResultTable.cs

```diff
@@ -94,6 +94,30 @@ public class ResultTable : IResultTable
     }
   }
 
+  async Task<Core.Common.Storage.Result> IResultTable.GetResult(string            sessionId,
+                                                                string            key,
+                                                                CancellationToken cancellationToken)
+  {
+    using var activity = activitySource_.StartActivity($"{nameof(IResultTable.GetResult)}");
+    activity?.SetTag($"{nameof(IResultTable.GetResult)}_sessionId",
+                     sessionId);
+    var sessionHandle    = sessionProvider_.Get();
+    var resultCollection = resultCollectionProvider_.Get();
+    try
+    {
+      return await resultCollection.AsQueryable(sessionHandle)
+                                   .Where(model => model.Id == Result.GenerateId(sessionId,
+                                                                                 key))
+                                   .SingleAsync(cancellationToken)
```

Comment:
This request is very different than the one from GetResults (use `model.Id`). Does using this approach in the `GetResults` approach lead to better performances (better usage of indexes) ?

Reply:
We can have a look later. But the performance of the batched version seems good, so I'm not worried about that. The goal of this PR was to fix a regression.